src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyPointcutParameters.java

```diff
@@ -6,17 +6,37 @@
  */
 package org.mule.extension.http.api.policy;
 
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+
 import org.mule.runtime.api.component.Component;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.util.MultiMap;
+import org.mule.runtime.core.api.util.ClassUtils;
 import org.mule.runtime.http.policy.api.HttpPolicyPointcutParameters;
 import org.mule.runtime.policy.api.PolicyPointcutParameters;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
 /**
  * Specific implementation of {@link PolicyPointcutParameters} for http:listener operation.
  * 
  * @since 1.0
  */
 public class HttpListenerPolicyPointcutParameters extends HttpPolicyPointcutParameters {
 
+  /**
+   * As long as the connector is compatible with runtime versions < 4.3.0, headers and masked request path need to be set using
+   * reflection since they were added in the API in 4.3.0
+   */
+  private static Method setHeadersMethod;
+  private static Method setMaskedRequestPathMethod;
+
+  static {
+    setHeadersMethod = ClassUtils.getMethod(HttpPolicyPointcutParameters.class, "setHeaders", null);
```

Comment:
what if this is running with a runtime version 4.2.x or 4.1.x?
Isn't this missing the parameter type?